### ISSUE TITLE
Datagrid Accessibility Defects

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3349]
The Datagrid was reviewed by Chris from the a11y team and he summarized his list of defects. This epic tracks all of the issues at a high level, some which require work, some which may be addressed by other in progress word, and some deprecations.

### Fixes required
* [ ] (160) Datagrids do not support arrow key navigation
* [ ] (1) Grid columns cannot be resized using the keyboard alone (previously known defect, need to design solution) #3372
* [x] (4) "Data Grid - Pagination", focus does not move to the beginning of the newly displayed data after the "clr-page-size-select" control has been activated (focus management not logical) #3370
* [x] (5) Data Grid - Pagination - focus is lost after activating either the "pagination-first" or "pagination-previous" buttons (focus management not logical) #3370
* [x] (6) Focus does not return to the datagrid-filter-toggle button after the datagrid-filter is closed (focus management not logical) #3370
* [x] (8) Screen reader users will not be aware of actions available from the clr-dg-action-overflow component when it appears (no announcing due to lack of context about button name) #3371
* [x] (9) There is no way for the developer using Clarity to provide an accessible name for the datagrid-action-toggle button. (should be done through common strings) #3371
* [x] (11) The clr-dg-cell should use role="gridcell" rather than role="cell" (improper role that needs review) #3373 
* [x] (15) The toggle button inside the clr-dg-column-toggle component lacks a mechanism for providing an accessible name (button name problem again) #3371
* [x] (17) Focus is lost after activating the "Select All" button in the clr-dg-column-toggle component (focus management) #3370
* [x] (19) The close button in the clr-dg-column-toggle component lacks a mechanism for providing an accessible name (button name again) #3371
* [ ] (168)The datagrid-filter-toggle button lacks an accessible name and state

### Should fix but not high severity

* [ ] (18) Screen reader users do not hear any status updates after the checkboxes in the clr-checkbox-wrapper components are activated (not indicating the status change of a selection choice)
* [x] (14) The first column in the clr-datagrid component is empty when using the clr-dg-row-detail component (should provide offscreen text instead of a blank column title) 

### Need retested after feature completion

* [x] (2) The "datagrid-filter" does not stay open after being activated with the keyboard (should be resolved by #2923)
* [x] (7) The clr-dropdown-menu component does not support arrow keys to select options (should be resolved by #3283)
* [x] (16) Focus is lost after activating the clr-dg-column-toggle (may be resolved by #2923)

### Deprecations, documentation, or expected to be resolved by other refactors 

* [x] (3) The clr-dg-filter component does not provide a mechanism to move focus to an element within a custom filter (should be replaced by new filtering pattern, can also be documented how a custom filter can do this itself) #2923 
* [x] (10) The contents of the clr-dg-row-detail component are not read by Mac OS X/VoiceOver (deprecation or major overhaul) #3375
* [x] (12) The "datagrid-expandable-caret-button" button does not indicate its expanded state to assistive technologies (deprecation or simple fix) #3375
* [ ] (13) The "datagrid-expandable-caret-button" button cannot be read or activated when using VoiceOver (deprecation or simple fix) #3375
* [x] Row actions popover is not accessible #3280

**Comment by clane** [comment]
Another solution for (4) would be to use a polite live region to announce a status update saying that the grid page has changed and to leave focus on the pagination button. This way we don't have to choose an element to focus and screen reader users will hear that the page has changed and hopefully understand that they need to go back and read the beginning of the grid.

**Comment by gnomeontherun** [comment]
Closing this epic, as all items have a ticket and this is just adding noise.
